drivers/flash/flash_ambiq.c

```diff
@@ -118,64 +118,17 @@ static int flash_ambiq_write(const struct device *dev, off_t offset, const void
 	return ret;
 }
 
-static int flash_ambiq_erase(const struct device *dev, off_t offset, size_t len)
-{
-	ARG_UNUSED(dev);
-
-	int ret = 0;
-
-	if (!flash_ambiq_valid_range(offset, len)) {
-		return -EINVAL;
-	}
-
-	/* The erase address and length alignment check will be done in HAL.*/
-
-	if (len == 0) {
-		return 0;
-	}
-
-	FLASH_SEM_TAKE();
-
-	ret = am_hal_mram_main_fill(AM_HAL_MRAM_PROGRAM_KEY, FLASH_ERASE_WORD,
-				    (uint32_t *)(SOC_NV_FLASH_ADDR + offset),
-				    (len / sizeof(uint32_t)));
-
-	FLASH_SEM_GIVE();
-
-	return ret;
-}
-
 static const struct flash_parameters *flash_ambiq_get_parameters(const struct device *dev)
 {
 	ARG_UNUSED(dev);
 
 	return &flash_ambiq_parameters;
 }
 
-#if CONFIG_FLASH_PAGE_LAYOUT
-static const struct flash_pages_layout pages_layout = {
-	.pages_count = SOC_NV_FLASH_SIZE / FLASH_ERASE_BLOCK_SIZE,
-	.pages_size = FLASH_ERASE_BLOCK_SIZE,
-};
-
-static void flash_ambiq_pages_layout(const struct device *dev,
-				     const struct flash_pages_layout **layout, size_t *layout_size)
-{
-	ARG_UNUSED(dev);
-
-	*layout = &pages_layout;
-	*layout_size = 1;
-}
-#endif /* CONFIG_FLASH_PAGE_LAYOUT */
-
 static const struct flash_driver_api flash_ambiq_driver_api = {
 	.read = flash_ambiq_read,
 	.write = flash_ambiq_write,
-	.erase = flash_ambiq_erase,
 	.get_parameters = flash_ambiq_get_parameters,
-#ifdef CONFIG_FLASH_PAGE_LAYOUT
-	.page_layout = flash_ambiq_pages_layout,
-#endif
```

Comment:
Hi @aaronyegx, @RichardSWheatley - could you guys take a look at these changes in the context of the Ambiq flash driver? Thanks :)

Reply:
We use this flash driver for Apollo4 MRAM and plan to add Apollo3 flash support which needs to erase before writing. It seems it will become incompatible after this change?

Reply:
That would be different device, so for the Apollo3  you can compile in the Flash and for Appollo4 you can compile the MRAM. Do you have any device that have both and you want to cover it with a single driver?

Reply:
Why was this removed from the Ambiq driver but not the Andes driver.

That Kconfig still has the page layout but ours has it removed?

Reply:
@de-nordic If we have added CONFIG_FLASH_HAS_EXPLICIT_ERASE to define if the device needs explicit erase or not, should we use this CONFIG_FLASH_HAS_EXPLICIT_ERASE to set the **explicit_erase** and include the flash_erase implementation instead of cleaning them up? As mentioned in my last comment, we will have Apollo3 flash support which need erase action soon and the erase driver (may only include some adaption) is still needed.

Reply:
Driver implementation  is up to you.
The flag is used by software to determine whether it has to do erase as required by the device itself, or sometimes use it for removal of data.
Most software  plays better with things that do not require erase.
I will pull the change here to separate PR so we can discuss your options separately.

Reply:
> Why was this removed from the Ambiq driver but not the Andes driver.
> 
> That Kconfig still has the page layout but ours has it removed?

That is why I need help from vendor representatives here. I do not have to remove these things, the question is how you want to approach it. You can still present MRAM/RRAM devices as Flash, they will just work less efficiently in software stack as it may still call erase even if not needed by device.
because the PR here prepares the subsystems to work with devices that have the erase requirement or not, change to any mram/rram can be done post this pr, in either direction, and subsystems will adapt.